### DISCUSSION TITLE
How to use QMarkupTable component within the UMD version?

### DISCUSSION THREAD

**Discussion by rubymongoose** [quasarframework/quasar#16994] [General - Components / Directives / etc]
Hello, I'm trying to figured out how to use QMarkupTable component within the UMD version of Quasar.

I saw the note for _UMD developers_ on component page.
And after some experiments with generated UMD file from here https://quasar.dev/start/umd
and playground codepen https://codepen.quasar.dev
I couldn't make QMarkupTable work.

But then I found this https://codepen.io/smolinari/pen/vYOmdyp
where QMarkupTable works fine and in Pen Settings it use quasar.umd.min.js file.

Question is - how to make QMarkupTable work in generated UMD file from here https://quasar.dev/start/umd
or in playground codepen https://codepen.quasar.dev 🤔

**Comment by yusufkandemir** [comment]
The idea is to use `<q-markup-table>` somewhere other than UMD, then use the developer tools to use the generated HTML code instead.

So, for just `<q-markup-table>`, you would need to use the following:
```html
<div
  class="q-markup-table q-table__container q-table__card q-table--horizontal-separator q-table--dark q-table__card--dark q-dark q-table--no-wrap"
>
  <table class="q-table">
    ...
  </table>
</div>
```
For `<q-markup-table separator="vertical">, you would need to use the following:
```html
<div
  class="q-markup-table q-table__container q-table__card q-table--vertical-separator q-table--dark q-table__card--dark q-dark q-table--no-wrap"
>
  <table class="q-table">
    ...
  </table>
</div>
```
and so on.

**Comment by rstoenescu** [comment]
Exactly how Scott's example is. https://codepen.io/smolinari/pen/vYOmdyp
Put the `<template>` there.

The reason QMarkupTable does not works out of the box without that is that the DOM is interpreted (and auto-fixed, especially on the `<table>` elements) by the browser before Vue can get to it and compile it.

**Reply by yusufkandemir** [reply]
If this is the "new" recommended solution, then the warning here needs to be updated:
https://quasar.dev/vue-components/markup-table#usage
![image](https://github.com/quasarframework/quasar/assets/6266078/a029a4e2-b7f8-49f5-ab0e-8b166afd9100)

**Reply by rstoenescu** [reply]
Did it.

**Reply by rubymongoose** [reply]
> Exactly how Scott's example is. https://codepen.io/smolinari/pen/vYOmdyp Put the `<template>` there.
> 
> The reason QMarkupTable does not works out of the box without that is that the DOM is interpreted (and auto-fixed, especially on the `<table>` elements) by the browser before Vue can get to it and compile it.

Thanks for the answer, but I still can't get this to work in playground https://codepen.quasar.dev
I think it's because example here https://codepen.io/smolinari/pen/vYOmdyp uses:

_quasar@1.9.4/dist/quasar.min.css
vue/dist/vue.js
quasar@1.9.4/dist/quasar.umd.min.js_

And current playground https://codepen.quasar.dev uses:

_quasar@2/dist/quasar.prod.css
vue@3/dist/vue.global.prod.js
quasar@2/dist/quasar.umd.prod.js_

In my case, table not showing and page contain this code:
```
<template>
	#document-fragment
</template>
```
